microservices/external_data_query/external_data_query/rcsb_pdb/api_models.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import dataclasses
+from typing import List, Optional, Union, Dict, Any
+
+from pydantic import BaseModel
+
+from external_data_query.mixins import BaseModelMixin
+
+# RCSB PDB
+
+@dataclasses.dataclass
+class FetchedProtein:
+    fasta_contents: str
+    link: str
+
+@dataclasses.dataclass
+class GetFastaFilesByIdsRequest(BaseModelMixin):
+    rcsb_pdb_ids: List[str]
+    job_id: Optional[str] = None
+
+@dataclasses.dataclass
+class GetFastaFilesResponse(BaseModelMixin):
+    fasta_contents: List[FetchedProtein]
+
+@dataclasses.dataclass
+class GetFastaFilesBySearchQueryRequest(BaseModelMixin):
+    search_query: str
+    max_results: int = None
+    exact_match: bool = False  # Added to support exact match queries
+    job_id: Optional[str] = None
+
+@dataclasses.dataclass
+class SequenceQueryRequest(BaseModelMixin):
+    sequence: str
+    sequence_type: str  # "protein", "dna", "rna"
```

Comment:
enum

Reply:
I don't think we can use enums as types in requests since it may confuse FastAPI but feel free to correct me if I'm wrong